### PR TITLE
net/gnrc/udp: Adding Null pointer check in gnrc_netif_hdr_build to avoid crashing of the node

### PR DESCRIPTION
**Pull request by Harisanthosh**
### Contribution description
This PR contains the fix for issue #11631  :

1. Check for Null pointer has been added in **tests/gnrc_udp/udp.c** file where the node was crashing when tried sending a payload over 5992 bytes MTU. By checking the return value of the function gnrc_netif_hdr_build before sending the payload to the destination address we avoid the Null pointer Dereferencing.

### Testing procedure

The fix has been tested in "native"
1.  Have a udps server running and make a note of it's IPv6 address
2.  From the **tests/gnrc_udp/** folder, build and run the program and try sending a large payload in the following format:  

> udp send [dest_addr] [port] [payload in bytes]

`udp send fe80::2806:1aff:fedd:f6d8 8888 5993`

3. The node will not send the message but will return an error message
`Error: unable to allocate NETIF header`

### Issues/PRs references

See issue: #11631.

### CODE REVIEW

File: tests/gnrc_udp/udp.c [hunk 1]
```diff
@@ -87,7 +87,6 @@ static void send(char *addr_str, char *port_str, char *data_len_str, unsigned in
     uint16_t port;
     ipv6_addr_t addr;
     size_t data_len;
-
```
Comment (jcarrano) [inline tests/gnrc_udp/udp.c]: Please leave this whitespace as is.

File: tests/gnrc_udp/udp.c [hunk 2]
```diff
@@ -137,7 +136,11 @@ static void send(char *addr_str, char *port_str, char *data_len_str, unsigned in
         /* add netif header, if interface was given */
         if (iface > 0) {
             gnrc_pktsnip_t *netif = gnrc_netif_hdr_build(NULL, 0, NULL, 0);
-
```
Comment (jcarrano) [inline tests/gnrc_udp/udp.c]: Same with this one.

File: tests/gnrc_udp/udp.c [hunk 3]
```diff
@@ -137,7 +136,11 @@ static void send(char *addr_str, char *port_str, char *data_len_str, unsigned in
...
+            if(netif == NULL) {
+                puts("Error: unable to allocate NETIF header");
+                gnrc_pktbuf_release(payload);
```
Comment (jcarrano) [inline tests/gnrc_udp/udp.c]: Shouldn't the `ip` buffer be released instead? (I'm not very familiar with how this works)

File: tests/gnrc_udp/udp.c [hunk 4]
```diff
@@ -256,3 +259,4 @@ int udp_cmd(int argc, char **argv)
     }
     return 0;
 }
+
```
Comment (jcarrano) [inline tests/gnrc_udp/udp.c]: You are introducing an additional newline here.

**Comment by Harisanthosh** [comment]
> Hi @Harisanthosh . Thanks for your contribution and welcome to the RIOT community.
> 
> I was able to reproduce the issue and I think your fix is OK. I'm not very knowledgeable on GNRC, but I'm not sure if it is the correct buffer that is being released.
> 
> Also, please rename your commit to comply with the format (e.g. "net/gnrc/udp: Add null pointer check in gnrc_netif_hdr_build").

File: tests/gnrc_udp/udp.c (hunk 1 above)
Comment (Harisanthosh) [reply tests/gnrc_udp/udp.c]: Thank you for your time and feedback @jcarrano ! I have made the necessary changes in the file, and update my commits as well!

File: tests/gnrc_udp/udp.c (hunk 3 above)
Comment (Harisanthosh) [reply tests/gnrc_udp/udp.c]: I tried releasing the IP buffer when I was experimenting and testing with multiple payloads. However, it resulted in Kernel Panic error. I believe resetting the payload cleared the way and allowed the resetting of UDP and IP, and thus the kernel panic error was avoided!
image

File: tests/gnrc_udp/udp.c (hunk 4 above)
Comment (Harisanthosh) [reply tests/gnrc_udp/udp.c]: Thank you for your prompt feedback, appreciate it very much. I have reopened the pull-request by inculcating your changes.

File: tests/gnrc_udp/udp.c (hunk 1 above)
Comment (Harisanthosh) [reply tests/gnrc_udp/udp.c]: I believe that I haven't used the static test tools to verify the quality of changes which resulted in whitespaces modifications. The change I made on top of it didn't resolve it, so I will make a fresh commit again after taking care of it ! Thanks

**Comment by Harisanthosh** [comment]
Closing this PR, owing to additional commits added due to incorrect usage of git rebase -i --autosquash. Will submit a new PR

**Comment by Harisanthosh** [comment]
New PR - https://github.com/RIOT-OS/RIOT/pull/12046

**Pull request closed without merging**